redis.go

```diff
@@ -168,15 +168,24 @@ func (s *RedisServer) Run(ctx context.Context) error {
 
 	// now start request/response workers and proxy calls and responses
 	pullArgs = append(pullArgs, redisPullTimeout) //the pull timeout
-	ch := s.Start(ctx, s.workers, s.cb)
+	workerCtx, shutdown := context.WithCancel(context.Background())
+	ch, wg := s.Start(workerCtx, s.workers, s.cb)
+
+	defer func() {
+		shutdown()
+		wg.Wait()
+		close(ch)
+	}()
+
 	for {
 		payload, err := s.getNext(pullArgs)
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
+
 		if err == redis.ErrNil {
+			select {
```

Comment:
```go
		select {
			case <-ctx.Done():
				return ctx.Err()
			default:
                                continue
			}
```
cleaner like this no? since there is nothing else in this block

Reply:
Yeah, it is cleaner but I try not use both `break` or `continue` inside select statements, because using a `break` gave me lots of issues before. A break inside a select actually breaks out of the `select` not from the loop. Probably continue does not have the same effect, but i would just avoid having it inside the select statement anyway.

Reply:
ha yes, we are in a loop, I didn't pay attention. ok got it.